test/blackbox-tests/test-cases/melange/simple.t/run.t

```diff
@@ -1,9 +1,8 @@
-Compilation using melange
-  $ dune build lib/.x.objs/melange/x.js
-  $ node ./_build/default/lib/.x.objs/melange/x.js
-  buy it
+Using `melange.emit` inside the same folder as the library will fail
 
-Rebuilding same project (js artifacts are tracked correctly)
-  $ dune build lib/.x.objs/melange/x.js
-  $ node ./_build/default/lib/.x.objs/melange/x.js
-  buy it
+  $ dune build lib/simple/x.js
+  Error: Multiple rules generated for _build/default/lib/.x.objs/x.ml.d:
```

Comment:
@rgrinberg I am not sure if this is the expected behavior, or this is a case we should support. If the latter, how would the output folder layout look like?

Reply:
this was fixed in https://github.com/ocaml/dune/pull/6349/commits/a9e1b706114b61173e716530f881e10341186eb1.